### PR TITLE
WIP move docker api types into their own package and scheme

### PR DESCRIPTION
**Pull request by smarterclayton**
Based on top of an earlier pull for demonstration

@deads2k this is what that move looks like

### CODE REVIEW

File: pkg/image/apis/image/docker/register.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+const (
+	GroupName       = "image.openshift.io"
+	LegacyGroupName = ""
+)
+
+var (
+	SchemeGroupVersion       = schema.GroupVersion{Group: GroupName, Version: runtime.APIVersionInternal}
```
Comment (deads2k) [inline pkg/image/apis/image/docker/register.go]: internal?

Comment (deads2k) [reply pkg/image/apis/image/docker/register.go]: > internal?

nm, wrong repo

File: pkg/image/apis/image/docker/scheme/scheme.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+var Scheme = runtime.NewScheme()
+var Codecs = serializer.NewCodecFactory(Scheme)
+var ParameterCodec = runtime.NewParameterCodec(Scheme)
+
+func init() {
+	AddToScheme(Scheme)
+}
+
+func AddToScheme(scheme *runtime.Scheme) {
```
Comment (deads2k) [inline pkg/image/apis/image/docker/scheme/scheme.go]: nit: call this `Install` to match the pull about to merge

File: pkg/image/apis/image/register.go [hunk 3]
```diff
@@ -54,6 +56,9 @@ func addLegacyKnownTypes(scheme *runtime.Scheme) error {
 		&ImageStreamTagList{},
 		&ImageStreamImage{},
 		&ImageStreamImport{},
+
+		// exposed for backwards compatibility
+		&docker.DockerImage{},
```
Comment (deads2k) [inline pkg/image/apis/image/register.go]: Move this into the register for the /docker package?

**Comment by deads2k** [comment]
yeah, I can live with that.

File: pkg/image/apis/image/register.go (hunk 3 above)
Comment (smarterclayton) [reply pkg/image/apis/image/register.go]: And then call the register from this method?  We need it in the v1 scheme, so do we want the /docker package adding things to a v1 group?

File: pkg/image/apis/image/docker/scheme/scheme.go (hunk 2 above)
Comment (smarterclayton) [reply pkg/image/apis/image/docker/scheme/scheme.go]: k

**Pull request closed without merging**